src/log.js

```diff
@@ -287,13 +287,13 @@ class Log extends GSet {
     // Update the internal entry index
     this._entryIndex = Object.assign(this._entryIndex, newItems)
 
+    // Update the length cache
+    this._length = Object.keys(this._entryIndex).length
```

Comment:
`Object.keys()` is an expensive operation as it'll have to "loop through" all the keys in `this._entryIndex`. As the size of the log keeps growing, it'll be slower and slower over time, ie. `O(n)` and I don't think we can do that.

A non-O(n) way to do this could be, for example, to increase the length as we loop through the `newItems` in https://github.com/orbitdb/ipfs-log/pull/190/files#diff-4f24bb7c46eda8f63690cb849ddb0f2dR295.

Any thoughts how to keep the length calculation not go through all items in the log but just the new ones?